lib/navigation.js

```diff
@@ -0,0 +1,51 @@
+// Navgation metalsmith plugin
+// builds a navigation object based on config file and folder structure
+
+// files: array containing information about every page
+// metalsmith: object containing global information such as meta data
+// done: function which must be called when the plugin has finished working
+
+const fs = require('fs')
+const path = require('path')
+const paths = require('../config/paths.json')
+const navigation = require('../config/navigation.json')
+
+module.exports = function () {
+  return function (files, metalsmith, done) {
+    // iterate main navigation item defined in navigation.json
+    for (let item in navigation) {
+      // if is not homepage look for subitems
+      if (navigation[item].url !== '') {
+        // define source path
+        let itemPath = paths.source + navigation[item].url
+        // get directories under main navigation item (e.g. ['breadcrumbs', 'checkboxes', ...])
+        let directories = fs.readdirSync(itemPath).filter(filePath => fs.statSync(path.join(itemPath, filePath)).isDirectory())
+        // if we have directories convert them into menu items
+        if (directories) {
+          // initialise child navigation items as array
+          navigation[item].items = []
+          // convert directory into a navigation item adding url, label and theme
+          for (let dir in directories) {
+            let url = navigation[item].url + '/' + directories[dir]
+            let frontmatter = files[url + '/index.html']
```

Comment:
This could maybe have an error check to check if that file exists, currently the build fails if it doesn't.

Reply:
The file exists for sure, that's how we ended up having it in the `files` array. I added a check for the frontmatter to exists.